### PR TITLE
Add methods to lock and unlock app.

### PR DESCRIPTION
**Pull request by gwuah**

### CODE REVIEW

File: resource_apps.go [hunk 1]
```diff
@@ -575,3 +575,50 @@ func (client *Client) AppNameAvailable(ctx context.Context, appName string) (boo
...
+func (client *Client) UnlockApp(ctx context.Context, input AppLockInput) (*App, error) {
+	query := `
+		mutation($input: AppLockInput!) {
+			lockApp(input: $input) {
+				app {
+					name
+				}
+			}
+		}
+	`
```
Comment (alichay) [inline resource_apps.go:611]: Is this the right mutation?

File: resource_apps.go [hunk 2]
```diff
@@ -575,3 +575,50 @@ func (client *Client) AppNameAvailable(ctx context.Context, appName string) (boo
 
 	return data.AppNameAvailable, nil
 }
+
+func (client *Client) LockApp(ctx context.Context, input CreateAppInput) (*AppLock, error) {
```
Comment (alichay) [inline resource_apps.go:579]: That type should probably be `AppLockInput` instead of `CreateAppInput`

**Review by alichay — CHANGES_REQUESTED** [review]

**Review by billyb2 — APPROVED** [review]
LGTM, though
I worry slightly about having app locks alongside machine leases. Will app locks clear automatically, like how leases do for machines? Also, what exactly will app locks be for?

**Comment by gwuah** [comment]
@billyb2 
app locks are different from machine locks, and the reason why we're bringing app locks is for https://github.com/superfly/flyctl/pull/3906

yes, they clear automatically, but we will also provide users a failsafe mechanism to unlock their apps, if things really really go south and they need to deploy.

I'm also going to make use of your feature flagging work to slowly release this to users, so we can monitor it. Either way, we need a way to prevent concurrent deployments.

**Comment by billyb2** [comment]
> @billyb2 app locks are different from machine locks, and the reason why we're bringing app locks is for [superfly/flyctl#3906](https://github.com/superfly/flyctl/pull/3906)
> 
> yes, they clear automatically, but we will also provide users a failsafe mechanism to unlock their apps, if things really really go south and they need to deploy.
> 
> I'm also going to make use of your feature flagging work to slowly release this to users, so we can monitor it. Either way, we need a way to prevent concurrent deployments.

makes sense 👍🏾

**Pull request closed without merging**